api/manifest.go

```diff
@@ -58,9 +62,42 @@ func getManifestForSHA(ctx context.Context, sha string) (manifest []byte, err er
 		Token:   &token,
 		Context: ctx,
 	}
-	return getGitHubReleaseAssetForSHA(&client, sha)
+	return loadOrFetchManifestForSHA(ctx, &client, sha)
 }
 
+// loadOrFetchManifestForSHA gets the bytes for the SHA's release's manifest json asset (unzipped).
+// The gzipped Value is stored in / loaded from memcache, to avoid unnecessary round-trips.
+func loadOrFetchManifestForSHA(ctx context.Context, client gitHubClient, sha string) (manifest []byte, err error) {
+	var body []byte
+	cached, err := memcache.Get(ctx, sha)
+	if err != nil && err != memcache.ErrCacheMiss {
+		return nil, err
+	} else if cached != nil {
+		body = cached.Value
+	} else {
+		if body, err = getGitHubReleaseAssetForSHA(client, sha); err != nil {
+			return nil, err
+		}
+		item := &memcache.Item{
+			Key:   sha,
+			Value: body,
+		}
+		// Shorter expiry for latest SHA, to keep it current.
```

Comment:
Shall we cache latest at all? If we do want to cache it, 20 minutes seem a bit too long (at least based on my observation during North American work hours). I'd suggest 5 minutes or so.

Reply:
Changed to 5 min.